<launch>
  <arg name="adapter_output" default="screen" />
  <arg name="relay_output" default="log" />
  <arg name="status_pub_hz" default="5.0" />
  <arg name="stop_reason_timeout" default="0.5" />
  <arg name="stop_reason_thresh_dist" default="100.0" />

  <arg name="input_steer" default="/vehicle/status/steering_status" />
  <arg name="input_vehicle_cmd" default="/control/command/control_cmd" />
  <arg name="input_turn_indicators" default="/vehicle/status/turn_indicators_status" />
  <arg name="input_hazard_lights" default="/vehicle/status/hazard_lights_status" />
  <arg name="input_odometry" default="/localization/kinematic_state" />
  <arg name="input_gear" default="/vehicle/status/gear_status" />
  <arg name="input_battery" default="/vehicle/status/battery_charge" />
  <arg name="input_nav_sat" default="/sensing/gnss/ublox/nav_sat_fix" />
  <arg name="input_autoware_state" default="/autoware/state" />
  <arg name="input_control_mode" default="/vehicle/status/control_mode" />
  <arg name="input_gate_mode" default="/control/current_gate_mode" />
  <arg name="input_emergency_state" default="/system/emergency/emergency_state" />
  <arg name="input_hazard_status" default="/system/emergency/hazard_status" />
  <arg name="input_stop_reason" default="/planning/scenario_planning/status/stop_reasons" />
  <arg name="input_v2x_command" default="/planning/scenario_planning/status/infrastructure_commands" />
  <arg name="input_v2x_state" default="/system/v2x/virtual_traffic_light_states" />
  <arg name="input_diagnostics" default="/diagnostics_agg" />
  <arg name="input_global_rpt" default="/pacmod/global_rpt" />
  <arg name="input_lane_change_available" default="/planning/scenario_planning/lane_driving/lane_change_available" />
  <arg name="input_lane_change_ready" default="/planning/scenario_planning/lane_driving/lane_change_ready" />
  <arg name="input_lane_change_candidate_path" default="/planning/scenario_planning/lane_driving/lane_change_candidate_path" />
  <arg name="input_path_change_ready" default="/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/ready_module" />
  <arg name="input_path_change_force_available" default="/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/force_available" />
  <arg name="input_path_change_running" default="/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/running_modules" />
  <arg name="input_obstacle_avoid_ready" default="/planning/scenario_planning/lane_driving/obstacle_avoidance_ready" />
  <arg name="input_obstacle_avoid_candidate_path" default="/planning/scenario_planning/lane_driving/obstacle_avoidance_candidate_trajectory" />
  <arg name="input_route" default="/planning/mission_planning/route" />
  <arg name="input_object" default="/perception/object_recognition/objects" />
  <arg name="input_traffic_signals" default="/perception/traffic_light_recognition/traffic_signals" />
  <arg name="input_nearest_traffic_signal" default="/planning/scenario_planning/lane_driving/behavior_planning/debug/traffic_signal" />
  <arg name="input_trajectory" default="/planning/scenario_planning/trajectory" />
  <arg name="input_current_max_velocity" default="/planning/scenario_planning/current_max_velocity" />
  <arg name="input_door_status" default="/pacmod/rear_pass_door_rpt" />
  <arg name="input_stop_speed_exceeded" default="/planning/scenario_planning/motion_velocity_smoother/stop_speed_exceeded" />
  <arg name="input_external_crosswalk_status" default="/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/input/external_crosswalk_states" />
  <arg name="input_external_intersection_status" default="/planning/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/input/external_intersection_states" />
  <arg name="input_expand_stop_range" default="/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/input/expand_stop_range" />
  <arg name="input_pose_initialization_request" default="/localization/util/pose_initialization_request" />

  <arg name="output_vehicle_status" default="vehicle/get/status" />
  <arg name="output_autoware_status" default="autoware/get/status" />
  <arg name="output_lane_change_status" default="lane_change/get/status" />
  <arg name="output_obstacle_avoid_status" default="object_avoidance/get/status" />
  <arg name="output_max_velocity" default="/planning/scenario_planning/max_velocity_default" />
  <arg name="output_autoware_engage" default="/autoware/engage" />
  <arg name="output_vehicle_engage" default="/vehicle/engage" />
  <arg name="output_goal" default="/planning/mission_planning/goal" />
  <arg name="output_route" default="/planning/mission_planning/route" />
  <arg name="output_lane_change_approval" default="/planning/scenario_planning/lane_driving/lane_change_approval" />
  <arg name="output_force_lane_change" default="/planning/scenario_planning/lane_driving/force_lane_change" />
  <arg name="output_path_change_approval" default="/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/path_change_approval" />
  <arg name="output_path_change_force" default="/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/path_change_force" />
  <arg name="output_obstacle_avoid_approval" default="/planning/scenario_planning/lane_driving/obstacle_avoidance_approval" />
  <arg name="output_door_control" default="/pacmod/rear_pass_door_cmd" />
  <!-- output_force_obstacle_avoid is not implemented -->
  <arg name="output_force_obstacle_avoid" default="tmp" />
  <arg name="output_overwrite_traffic_signals" default="/external/traffic_light_recognition/traffic_signals" />

  <arg name="node_emergency_stop" default="/control/vehicle_cmd_gate" />
  <arg name="param_emergency_stop" default="use_external_emergency_stop" />
  <arg name="node_max_velocity" default="/planning/scenario_planning/motion_velocity_smoother" />
  <arg name="param_max_velocity" default="max_velocity" />

  <arg name="set_max_velocity" default="vehicle/put/velocity" />
  <arg name="set_temporary_stop" default="vehicle/put/stop" />
  <arg name="set_engage" default="autoware/put/engage" />
  <arg name="set_route" default="autoware/put/route" />
  <arg name="set_goal" default="autoware/put/goal" />
  <arg name="set_lane_change_approval" default="lane_change/put/approval" />
  <arg name="set_force_lane_change" default="lane_change/put/force" />
  <arg name="set_path_change_approval" default="path_change/put/approval" />
  <arg name="set_path_change_force" default="path_change/put/force" />
  <arg name="set_obstacle_avoid_approval" default="object_avoidance/put/approval" />
  <arg name="set_force_obstacle_avoid" default="object_avoidance/put/force" />
  <arg name="set_overwrite_traffic_signals" default="traffic_light/put/traffic_signals" />
  <arg name="set_door_control" default="vehicle/put/door" />
  <arg name="set_crosswalk_status" default="autoware/put/crosswalk_states" />
  <arg name="set_intersection_status" default="autoware/put/intersection_states" />
  <arg name="set_expand_stop_range" default="autoware/put/expand_stop_range" />
  <arg name="set_pose_initialization_request" default="autoware/put/pose_initialization_request" />

  <arg name="get_route" default="autoware/get/route" />
  <arg name="get_predicted_object" default="prediction/get/objects" />
  <arg name="get_door_status" default="vehicle/get/door" />
  <arg name="get_stop_speed_exceeded" default="autoware/get/stop_speed_exceeded" />
  <arg name="get_traffic_signals" default="traffic_light/get/traffic_signals" />
  <arg name="get_nearest_traffic_signal" default="traffic_light/get/nearest_traffic_signal" />
  <arg name="get_v2x_command" default="/awapi/tmp/infrastructure_commands" />
  <arg name="get_v2x_state" default="/awapi/tmp/virtual_traffic_light_states" />
  <arg name="get_path_change_ready" default="path_change/get/ready_module" />
  <arg name="get_path_change_force_available" default="path_change/get/force_available" />
  <arg name="get_path_change_running" default="path_change/get/running_modules" />

  <arg name="use_intra_process" default="false" />
  <arg name="use_multithread" default="false" />

  <group>
    <push-ros-namespace namespace="awapi"/>
    <node pkg="awapi_awiv_adapter" exec="awapi_awiv_adapter" output="screen">
      <remap from="input/steer" to="$(var input_steer)" />
      <remap from="input/vehicle_cmd" to="$(var input_vehicle_cmd)" />
      <remap from="input/turn_indicators" to="$(var input_turn_indicators)" />
      <remap from="input/hazard_lights" to="$(var input_hazard_lights)" />
      <remap from="input/odometry" to="$(var input_odometry)" />
      <remap from="input/gear" to="$(var input_gear)" />
      <remap from="input/battery" to="$(var input_battery)" />
      <remap from="input/nav_sat" to="$(var input_nav_sat)" />
      <remap from="input/autoware_state" to="$(var input_autoware_state)" />
      <remap from="input/control_mode" to="$(var input_control_mode)" />
      <remap from="input/gate_mode" to="$(var input_gate_mode)" />
      <remap from="input/emergency_state" to="$(var input_emergency_state)" />
      <remap from="input/hazard_status" to="$(var input_hazard_status)" />
      <remap from="input/stop_reason" to="$(var input_stop_reason)" />
      <remap from="input/v2x_command" to="$(var input_v2x_command)" />
      <remap from="input/v2x_state" to="$(var input_v2x_state)" />
      <remap from="input/diagnostics" to="$(var input_diagnostics)" />
      <remap from="input/global_rpt" to="$(var input_global_rpt)" />
      <remap from="input/lane_change_available" to="$(var input_lane_change_available)" />
      <remap from="input/lane_change_ready" to="$(var input_lane_change_ready)" />
      <remap from="input/lane_change_candidate_path" to="$(var input_lane_change_candidate_path)" />
      <remap from="input/obstacle_avoid_ready" to="$(var input_obstacle_avoid_ready)" />
      <remap from="input/obstacle_avoid_candidate_path" to="$(var input_obstacle_avoid_candidate_path)" />
      <remap from="input/max_velocity" to="$(var set_max_velocity)" />
      <remap from="input/current_max_velocity" to="$(var input_current_max_velocity)" />
      <remap from="input/temporary_stop" to="$(var set_temporary_stop)" />
      <remap from="input/autoware_trajectory" to="$(var input_trajectory)" />
      <remap from="input/door_control" to="$(var set_door_control)" />
      <remap from="input/door_status" to="$(var input_door_status)" />
      <remap from="output/vehicle_status" to="$(var output_vehicle_status)" />
      <remap from="output/autoware_status" to="$(var output_autoware_status)" />
      <remap from="output/lane_change_status" to="$(var output_lane_change_status)" />
      <remap from="output/obstacle_avoid_status" to="$(var output_obstacle_avoid_status)" />
      <remap from="output/max_velocity" to="$(var output_max_velocity)" />
      <remap from="output/door_control" to="$(var output_door_control)" />
      <remap from="output/door_status" to="$(var get_door_status)" />
      <remap from="output/v2x_command" to="$(var get_v2x_command)" />
      <remap from="output/v2x_state" to="$(var get_v2x_state)" />
      <param name="node/emergency_stop" value="$(var node_emergency_stop)" />
      <param name="param/emergency_stop" value="$(var param_emergency_stop)" />
      <param name="node/max_velocity" value="$(var node_max_velocity)" />
      <param name="param/max_velocity" value="$(var param_max_velocity)" />
      <param name="status_pub_hz" value="$(var status_pub_hz)" />
      <param name="stop_reason_timeout" value="$(var stop_reason_timeout)" />
      <param name="stop_reason_thresh_dist" value="$(var stop_reason_thresh_dist)" />
    </node>
  </group>

  <include file="$(find-pkg-share awapi_awiv_adapter)/launch/awapi_relay_container.launch.py" />

</launch>
